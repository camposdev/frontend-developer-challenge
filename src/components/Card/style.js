import styled from "styled-components";

export const Wrapper = styled.div``;

export const Image = styled.img`
  display: block;
  width: 100%;
  border: 1px solid var(--secondary);
  margin-bottom: 15px;
`;

export const Title = styled.h4`
  font-size: 1.4rem;
  font-weight: normal;
  margin-bottom: 10px;
`;

export const Overview = styled.p`
  font-size: 1.2rem;
  margin: 0 0 10px;
`;

export const OldPrice = styled.div`
  font-size: 1.2rem;
  margin-bottom: 5px;
`;

export const Price = styled.div`
  font-size: 1.6rem;
  font-weight: bold;
  margin-bottom: 5px;
`;

export const Installments = styled.div`
  font-size: 1.2rem;
  margin-bottom: 10px;
`;