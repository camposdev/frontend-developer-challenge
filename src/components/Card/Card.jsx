import React from "react";
import PropTypes from "prop-types";
import numeral from "numeral";
import Button from "../Button";
import { Wrapper, Image, Title, Overview, OldPrice, Price, Installments } from "./style";

const Card = ({ product }) => (
  <Wrapper>
    <Image src={product.image} alt={product.name} />
    <Title>{product.name}</Title>
    <Overview>{product.description}</Overview>
    <OldPrice>De: {numeral(product.oldPrice).format('$0,0.00')}</OldPrice>
    <Price>Por: {numeral(product.price).format('$0,0.00')}</Price>
    <Installments>ou {product.installments.count}x de {numeral(product.installments.value).format('$0,0.00')}</Installments>
    <Button flat small block>Comprar</Button>
  </Wrapper>
);

Card.propTypes = {
  product: PropTypes.object.isRequired
}

export default Card;
