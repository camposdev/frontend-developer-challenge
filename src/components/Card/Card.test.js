import React from "react";
import renderer from "react-test-renderer";
import Card from "./Card";

const mock = {
  description: "Product long description number 0 just to make more than one like of text.",
  id: 0,
  image: "//imagens.pontofrio.com.br/Control/ArquivoExibir.aspx?IdArquivo=5783178",
  installments: {count: 10, value: 35.2},
  name: "Product Number 0",
  oldPrice: 398,
  price: 352
}

describe("Card test suit", () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Card product={mock} />)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});