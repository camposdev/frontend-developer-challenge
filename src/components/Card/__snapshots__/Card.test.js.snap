// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card test suit renders correctly 1`] = `
<div
  className="sc-bwzfXH uTxCW"
>
  <img
    alt="Product Number 0"
    className="sc-htpNat PQLaC"
    src="//imagens.pontofrio.com.br/Control/ArquivoExibir.aspx?IdArquivo=5783178"
  />
  <h4
    className="sc-bxivhb dkxgvx"
  >
    Product Number 0
  </h4>
  <p
    className="sc-ifAKCX efpgmk"
  >
    Product long description number 0 just to make more than one like of text.
  </p>
  <div
    className="sc-EHOje hmtkNb"
  >
    De: 
    $398.00
  </div>
  <div
    className="sc-bZQynM gbyGVL"
  >
    Por: 
    $352.00
  </div>
  <div
    className="sc-gzVnrw dHbKdU"
  >
    ou 
    10
    x de 
    $35.20
  </div>
  <button
    className="sc-bdVaJa dsxofe"
  >
    Comprar
  </button>
</div>
`;
